spyderlib/spyder_restart.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Restart Spyder
+
+A helper script to allow for spyder to restart
+"""
+
+import os
+import os.path as osp
+import sys
+import platform
+import subprocess
+import time
+
+from spyderlib.py3compat import to_text_string
+
+
+def main():
+    # Variables defined in spyderlib\spyder.py 'restart()' method
+    spyder_args = os.environ.pop('SPYDER_ARGS', None)
+    pid = os.environ.pop('SPYDER_PID', None)
+    is_bootstrap = os.environ.pop('SPYDER_IS_BOOTSTRAP', None)
+    spyder_folder = os.environ.pop('SPYDER_FOLDER', None)
+
+    if any([not spyder_args, not pid, not is_bootstrap, not spyder_folder]):
+        error = "This script can only be called from within a Spyder instance"
+        raise RuntimeError(error)
+
+    # Fix variables
+    args = eval(spyder_args)
+    args = ' '.join(args)
+    is_bootstrap = eval(is_bootstrap)
+    pid = eval(pid)
+
+    # Fix args
+    # Maybe --new-instance should always be included to cope with corner cases
+    # where someone has one running version without and and another with
+    # If the user the one without (--new...) it would not start
+
+    python = sys.executable
+
+    # Build the command
+    if is_bootstrap:
+        spyder = osp.join(spyder_folder, 'bootstrap.py')
+    else:
+        spyderlib = osp.join(spyder_folder, 'spyderlib')
+        spyder = osp.join(spyderlib, 'start_app.py')
+
+    command = '"{0}" "{1}" {2}'.format(python, spyder, args)
+
+    # Adjust the command depending on the OS
+    system = platform.system().lower()
+    if system.startswith('win'):
+        shell = False
+        is_pid_running = _is_pid_running_on_windows
+    elif system.startswith('linux'):
+        shell = True
+        is_pid_running = _is_pid_running_on_unix
+    elif system.startswith('darwin'):
+        shell = True
+        is_pid_running = _is_pid_running_on_unix
+
+    # Wait for original process to end before launching the new instance
+    while True:
+        if not is_pid_running(pid):
+            break
+        time.sleep(0.2)
+
+    env = os.environ.copy()
+    try:
+        subprocess.Popen(command, shell=shell, env=env)
+    except Exception as error:
+        print(command)
+        print(error)
+        time.sleep(15)
+
+
+def _is_pid_running_on_windows(pid):
```

Comment:
I think you did not read thoroughly what the functions are doing.

**Linux/OSX**
For linux I took the answer from [here](http://stackoverflow.com/questions/568271/how-to-check-if-there-exists-a-process-with-a-given-pid) so the process is not actually killed when you pass the argument `0` it is just poked. 

**Windows** 
On windows the previous method does not work, because in windows IT DOES actually kills the process. That is why the windows version is running a NEW subprocess and running the command [TASKLIST](http://ss64.com/nt/tasklist.html)... which list all process... I am just looking if the PID uis listed otherwise kill the process (not the windows one... but the one spawned to use TASKLIST)

This process is repeated with a timer to so the check is a bit spaced in time (works better) and only when then when it has detyected that the process has effectively ended, it will restart a new instance of spyder. 

In no part whatsoever I am killing or murdering brutally the previous instance of spyder...@nodd, @ccordoba12